airflow/providers/google/cloud/transfers/bigquery_to_postgres.py

```diff
@@ -0,0 +1,137 @@
+        ``[A,B,C]`` and you pass 'B,A' in the ``selected_fields``
+        the data would still be of the form ``'A,B'`` and passed through this form
+        to PostgreSQL
+
+    **Example**: ::
+
+       # [START howto_operator_bigquery_to_postgres]
+       transfer_data = BigQueryToPostgresOperator(
+            task_id='task_id',
+            dataset_table='origin_bq_table',
+            postgres_table='dest_table_name',
+            replace=True,
+        )
+        # [END howto_operator_bigquery_to_postgres]
+
+    :param dataset_table: A dotted ``<dataset>.<table>``: the big query table of origin
+    :param selected_fields: List of fields to return (comma-separated). If
+        unspecified, all fields are returned.
+    :param gcp_conn_id: reference to a specific Google Cloud hook.
+    :param postgres_conn_id: Reference to :ref:`postgres connection id <howto/connection:postgres>`.
+    :param database: name of database which overwrite defined one in connection
+    :param replace: Whether to replace instead of insert
+    :param batch_size: The number of rows to take in each batch
+    :param location: The location used for the operation.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    """
+
+    template_fields: Sequence[str] = (
+        "dataset_id",
+        "table_id",
+        "postgres_table",
+        "impersonation_chain",
+    )
+
+    def __init__(
+        self,
+        *,
+        dataset_table: str,
+        postgres_table: str,
+        selected_fields: list[str] | str | None = None,
+        gcp_conn_id: str = "google_cloud_default",
+        postgres_conn_id: str = "postgres_default",
+        database: str | None = None,
+        replace: bool = False,
+        batch_size: int = 1000,
+        location: str | None = None,
+        impersonation_chain: str | Sequence[str] | None = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.selected_fields = selected_fields
+        self.gcp_conn_id = gcp_conn_id
+        self.postgres_conn_id = postgres_conn_id
+        self.database = database
+        self.postgres_table = postgres_table
+        self.replace = replace
+        self.batch_size = batch_size
+        self.location = location
+        self.impersonation_chain = impersonation_chain
+        try:
+            self.dataset_id, self.table_id = dataset_table.split(".")
```

Comment:
I refactored the implementation. I am now using a `BaseBigQueryToSqlOperator`. In order to remain as backward compatible as possible, I used the old implementation tied to the `MySQL` operator.

If the user does not provide a correct input it will fail at runtime anyway (in the `execute` method).